### DISCUSSION TITLE
Typing issue: ASGITransport's app argument is not compatible with FastAPI/Starlette app.

### DISCUSSION THREAD

**Discussion by AlexanderPodorov** [encode/httpx#3104] [Potential Issue]
See example code:

```python
from fastapi import FastAPI
from httpx import ASGITransport, AsyncClient

# I guess the same issue will be for Starlette app, since FastAPI class just inherits from Starlette
app = FastAPI()


async def test() -> None:
    # line below shows typing issue by pyright:
    async with AsyncClient(base_url="http://test-server.com", transport=ASGITransport(app)) as client:
        await client.get("")

```
I'm not sure is it a proper fix, but replacing all `Dict` with `MutableMapping` right there solves the issue:
https://github.com/encode/httpx/blob/326b9431c761e1ef1e00b9f760d1f654c8db48c6/httpx/_transports/asgi.py#L19-L26

`httpx==0.27.0`
`fastapi==0.109.2`
`pyright==1.1.351`

**Comment by waketzheng** [comment]
Fixed by this one: https://github.com/encode/httpx/pull/3109
Pass mypy check after upgrade httpx:
```bash
pip install --upgrade 'httpx>=0.27.1' || pip install --upgrade --force-reinstall git+ssh://git@github.com/encode/httpx
```

**Comment by ugyballoons** [comment]
Is there any news when this latest update will be released?
`ERROR: No matching distribution found for httpx>=0.27.1`
